File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_cardaccf.py
"""Reference emit for CARDACCF — multi-account fee table → 100.00."""
# EXPECTED: 100.00
RATE_A = 0.0200
RATE_D = 0.0550
LATE_FEE = 25.00
ACCOUNTS = [
    ("A", 1000.00, 0),
    ("D", 1000.00, 45),
    ("C", 500.00, 0),
]
total = 0.0
for status, bal, days in ACCOUNTS:
    if status == "A":
        fee = round(bal * RATE_A + 1e-12, 2)
    elif status == "D":
        fee = round(bal * RATE_D + 1e-12, 2)
        if days > 30:
            fee = round(fee + LATE_FEE + 1e-12, 2)
    else:
        fee = 0.0
    total = round(total + fee + 1e-12, 2)
print(f"{total:.2f}")

File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_cardaccf.cs
// EXPECTED: 100.00
// Reference emit for CARDACCF — multi-account fee table (WHEN/IF/COMPUTE).
using System;
public static class CardaccfRef {
  public static void Main() {
    double rateA = 0.0200;
    double rateD = 0.0550;
    double lateFee = 25.00;
    (string status, double bal, int days)[] accounts = {
      ("A", 1000.00, 0),
      ("D", 1000.00, 45),
      ("C", 500.00, 0),
    };
    double total = 0.0;
    foreach (var (status, bal, days) in accounts) {
      double fee = status switch {
        "A" => Math.Round(bal * rateA + 1e-12, 2),
        "D" => Math.Round(bal * rateD + 1e-12, 2),
        _ => 0.0,
      };
      if (status == "D" && days > 30) {
        fee = Math.Round(fee + lateFee + 1e-12, 2);
      }
      total = Math.Round(total + fee + 1e-12, 2);
    }
    Console.WriteLine(total.ToString("0.00"));
  }
}
